### DISCUSSION TITLE
Etag restoration

### DISCUSSION THREAD

**Discussion by junukwon7** [hydro-dev/Hydro#52] [Q&A] [answered]
I have some TCs with no etags. I would like to fix it though cli because there are plenty of them.
![image](https://user-images.githubusercontent.com/48399106/107870400-6ec17f00-6edb-11eb-8c1d-373160ba01cd.png)

Where does hydro make the infos of TCs?

Plz help me.

https://github.com/hydro-dev/Hydro/blob/3553c2c3541082bebf5797a70891af8a37aee14a/packages/hydrooj/src/model/problem.ts#L151-L160

**Comment by junukwon7** [comment, marked as answer]
Solved it! Just made a function that only gets and sets the metadata.
```ts
async function setTestdata(domainId, pid, name) {
    const [[, fileinfo]] = await Promise.all([
        document.getSub(domainId, document.TYPE_PROBLEM, pid, 'data', name),
    ]);
    const meta = await storage_1.default.getMetaCli(`problem/${domainId}/${pid}/testdata/${name}`);
    const payload = { name, ...lodash_1.pick(meta, ['size', 'lastModified', 'etag']) };
    await push(domainId, pid, 'data', { _id: name, ...payload });
}
exports.setTestdata = setTestdata;
```
Like this.

**Reply by undefined-moe** [reply]
glad to see that :D
